assets/js/SignGroups.tsx

```diff
@@ -1,7 +1,361 @@
+  StationConfig,
+  Zone,
+  RouteSignGroups,
+  SignGroup,
+} from './types';
+import { defaultZoneLabel } from './helpers';
 
-function SignGroups(): JSX.Element | null {
-  return <div>Sign Groups Here</div>;
+interface ZoneSelectorProps {
+  config: StationConfig;
+  zone: Zone;
+  line: string;
+  selectedSigns: { [sign: string]: boolean };
+  onSignChange: (signId: string, selected: boolean) => void;
+  readOnly: boolean;
+  kind: 'default' | 'named';
+}
+
+function ZoneSelector({
+  config,
+  zone,
+  line,
+  selectedSigns,
+  onSignChange,
+  readOnly,
+  kind,
+}: ZoneSelectorProps): JSX.Element | null {
+  const handleChange = React.useCallback(
+    (evt) => {
+      if (!readOnly) {
+        onSignChange(evt.target.name, evt.target.checked);
+      }
+    },
+    [onSignChange, readOnly],
+  );
+
+  const zoneConfig = config.zones[zone];
+
+  if (
+    zoneConfig === undefined ||
+    (zoneConfig.label === undefined && kind === 'named') ||
+    (zoneConfig.label !== undefined && kind === 'default')
+  ) {
+    return null;
+  }
+
+  const zoneLabel = zoneConfig.label || defaultZoneLabel(zone);
+  const signId = arincToRealtimeId(`${config.id}-${zone}`, line);
+  const isSelected = selectedSigns[signId] || false;
+
+  return (
+    <div>
+      <label
+        className={`sign_groups--zone_selector${
+          isSelected ? ' sign_groups--zone_selector-selected' : ''
+        }`}
+      >
+        <input
+          type="checkbox"
+          name={signId}
+          checked={isSelected}
+          onChange={handleChange}
+        />
+        {zoneLabel}
+      </label>
+    </div>
+  );
+}
+
+interface SignGroupsFormProps {
+  line: string;
+  currentTime: number;
+  alerts: RouteAlerts;
+  setSignGroup: (line: string, ts: number, signGroup: SignGroup) => void;
+  readOnly: boolean;
+}
+
+function SignGroupsForm({
+  line,
+  currentTime,
+  alerts,
+  setSignGroup,
+  readOnly,
+}: SignGroupsFormProps): JSX.Element | null {
+  const [selectedSigns, setSelectedSigns] = React.useState<{
+    [sign: string]: boolean;
+  }>({});
+  const [line1, setLine1] = React.useState('');
+  const [line2, setLine2] = React.useState('');
+  const [expireDT, setExpireDT] = React.useState<Date | null>(null);
+  const [expireAlertID, setExpireAlertID] = React.useState('');
+
+  const onDatePickerChange = React.useCallback(
+    (date: Date | null) => {
+      setExpireDT(date);
+    },
+    [setExpireDT],
+  );
+
+  const onSignChange = React.useCallback(
+    (signId: string, isChecked: boolean) => {
+      setSelectedSigns({ ...selectedSigns, ...{ [signId]: isChecked } });
+    },
+    [selectedSigns, setSelectedSigns],
+  );
+
+  const handleSubmit = React.useCallback(
+    (event) => {
+      event.preventDefault();
+
+      /* eslint-disable @typescript-eslint/no-unused-vars -- the _id, and _isChecked */
+      const signIds = Object.entries(selectedSigns)
+        .filter(([_id, isChecked]) => isChecked)
+        .map(([id, _isChecked]) => id);
```

Comment:
I think you can do this, and then we shouldn't need to disable the warning:

```js
.filter(([, isChecked]) => isChecked)
.map(([id, ]) => id);
```

Reply:
Yeah, I had that, and it does appease eslint, but it looked pretty subtle to me, easy to miss the comma, especially for the `map` where in a lot of situations trailing commas are optional. And I liked indicating that the first argument in the `filter` was `_id` to make the chained `map` a little easier to follow.

But, thinking about it now, I don't want to "write Elixir in Javascript", so since this is how it's done, I should probably just get with the program and get used to this pattern.